docs/src/architecture/state.md

```diff
@@ -1,74 +1,86 @@
 # State
+The state of the network describes the current condition of all individual accounts and note states. It describes what is currently the case. With its state model, using concurrent offchain state, Polygon Miden aims to realise private transactions, and execution and state bloat minimization. 
 
-The state of the network describes the current condition of all individual accounts and note states. It describes what is currently the case.
+Privacy mostly results from UTXO-like state, having notes and nullifiers, and zero-knowledge proofs. Execution bloat happens when transactions get re-executed by all participants of the network. State bloat describes the ever growing state stored in blockchain nodes. Zero-knowledge proofs can only contribute partly to the solution, a concurrent state model with offchain state is required.
 
+| | Minimize execution bloat | Minimize state bloat |
+|---|---|---|
+| **Result of** | Verfifying state by re-executing | Storing all data onchain as state |
+| **Zero-knowledge helps with** | Transactions executed only once | No need to know full state to validate blocks |
+| **Concurrent state model helps with** | Transactions executed concurrently by distinct actors | No need to kow full state to produce blocks |
+
+
+## State components
 The Miden Node(s) maintain three databases to describe the state:
 1. A database of accounts.
 2. A database of notes.
 3. A database of nullifiers for already consumed notes.
 
 <p align="center">
-  <img src="../diagrams/architecture/state/State.png">
+  <img src="../diagrams/architecture/state/State.png" width="80%">
 </p>
 
-## State components
-
 These databases are represented by authenticated data structures, such that we can easily prove that items were added to or removed from a database, and a commitment to the database would be very small.
 
-### Account database
-Current account states are recorded in a Sparse Merkle tree which maps account IDs to account hashes, where the account hash is computed as
+Polygon Miden has two databases to capture the note states. The note database is append-only and permanently stores all notes, while the nullifier database stores which notes cannot be consumed anymore. Separating note storage into these two databases gives Polygon Miden client-side proving and advanced privacy.
 
-`hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`.
+### Account database
+The latest account states are recorded in a [tiered sparse Merkle tree](https://0xpolygonmiden.github.io/miden-base/crypto-primitives/tsmt.html) which maps account IDs to account hashes and account data if needed. 
```

Comment:
...and account data for on-chain accounts.

Reply:
done